### PR TITLE
chore(demo): replace all hard-coded documentation relative links with `DemoRoute`

### LINKED ISSUE #7186: 📚 - Replace all hard-coded documentation relative links with `DemoRoute`
### Description

1. Find all relative links
```html
<a
    routerLink="/relative-link/to/another/page"
    tuiLink
></a>
```
2. Replace hard-coded value with `DemoRoute`.

See example:
* https://github.com/taiga-family/taiga-ui/pull/7185

### PR DESCRIPTION
**Pull request by nsbarsukov**
Closes #7186

```ts
import type {Rule, Tree} from '@angular-devkit/schematics';
import {type DevkitFileSystem, getNgComponents, saveActiveProject, Scope} from 'ng-morph';

import {DemoRoute} from '../../../../demo/src/modules/app/demo-routes';
import {ALL_TS_FILES} from '../../constants';
import {addUniqueImport} from '../../utils/add-unique-import';
import {findElementsWithAttributeOnTag} from '../../utils/templates/elements';
import {getComponentTemplates} from '../../utils/templates/get-component-templates';
import {
    getPathFromTemplateResource,
    getTemplateFromTemplateResource,
    getTemplateOffset,
} from '../../utils/templates/template-resource';
import {getFileSystem} from '../utils/get-file-system';

const TAG_NAME = 'a';
const ATTR_NAME = 'routerLink';

function migrate(fileSystem: DevkitFileSystem): void {
    const componentWithTemplatesPaths = getComponentTemplates(ALL_TS_FILES);

    componentWithTemplatesPaths.forEach(resource => {
        const path = fileSystem.resolve(getPathFromTemplateResource(resource));
        const recorder = fileSystem.edit(path);

        const template = getTemplateFromTemplateResource(resource, fileSystem);
        const templateOffset = getTemplateOffset(resource);

        const elements = findElementsWithAttributeOnTag(
            template,
            [ATTR_NAME],
            [TAG_NAME],
            element =>
                Boolean(
                    element.attrs
                        .find(attr => attr.name === ATTR_NAME.toLowerCase())
                        ?.value.startsWith('/'),
                ),
        );

        elements.forEach(element => {
            const {name, value} =
                element.attrs.find(attr => ATTR_NAME.toLowerCase() === attr.name) || {};

            if (!name || !value) {
                return;
            }

            const {startOffset, endOffset} = element.sourceCodeLocation?.attrs?.[
                name
            ] || {startOffset: 0, endOffset: 0};
            const dictionaryKey =
                Object.entries(DemoRoute).find(
                    ([_, dictValue]) => dictValue === value,
                )?.[0] || 'TODO';

            recorder.remove(templateOffset + startOffset, endOffset - startOffset);
            recorder.insertRight(
                templateOffset + startOffset,
                `[${ATTR_NAME}]="docRoutes.${dictionaryKey}"`,
            );

            const [ngComponent] = getNgComponents(resource.componentPath);

            addUniqueImport(
                ngComponent.getSourceFile().getFilePath(),
                'DemoRoute',
                '@demo/routes',
            );

            !ngComponent.getProperty('docRoutes') &&
                ngComponent.addProperty({
                    name: 'docRoutes',
                    initializer: 'DemoRoute',
                    isReadonly: true,
                    scope: Scope.Protected,
                });
        });
    });
}

export function run(): Rule {
    return (tree: Tree) => {
        const fileSystem = getFileSystem(tree);

        migrate(fileSystem);

        fileSystem.commitEdits();
        saveActiveProject();
    };
}

```

### CODE REVIEW

File: projects/demo/src/modules/components/surface/index.html [hunk 1]
```diff
@@ -7,8 +7,8 @@
         <p>
             General purpose container used in Taiga UI interfaces. Often used in conjunction with
             <a
-                routerLink="/experimental/card"
                 tuiLink
+                [routerLink]="docRoutes.CardLarge"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/surface/index.html:11]:
```suggestion
                [routerLink]="docRoutes.CardMedium"
```

File: projects/demo/src/modules/components/surface/index.html [hunk 2]
```diff
@@ -43,8 +43,8 @@
                     <code>border-radius</code>
                     are not part of the surface. Take a look at
                     <a
-                        routerLink="/experimental/card-large"
                         tuiLink
+                        [routerLink]="docRoutes.CardMedium"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/surface/index.html:47]:
```suggestion
                        [routerLink]="docRoutes.CardLarge"
```

File: projects/demo/src/modules/components/confirm/index.ts [hunk 3]
```diff
@@ -29,6 +30,7 @@ export default class ExampleComponent implements TuiConfirmData {
     private readonly dialogs = inject(TuiDialogService);
     private readonly alerts = inject(TuiAlertService);
 
+    protected readonly docRoutes = DemoRoute;
```
Comment (splincode) [inline projects/demo/src/modules/components/confirm/index.ts:33]: why not just `protected readonly routes = DemoRoute;`?

File: projects/demo/src/modules/components/surface/index.html (hunk 1 above)
Comment (nsbarsukov) [inline projects/demo/src/modules/components/surface/index.html:11]:
```suggestion
                [routerLink]="docRoutes.CardMedium"
```

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/components/confirm/index.ts (hunk 3 above)
Comment (nsbarsukov) [reply projects/demo/src/modules/components/confirm/index.ts:33]: No problems, I'll rename it

Comment (nsbarsukov) [reply projects/demo/src/modules/components/confirm/index.ts]: Fixed in https://github.com/taiga-family/taiga-ui/pull/7642/commits/883751d9ff2f3ed444f7f1dc49498204c0b476ca

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**